hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/infra/HevmTransactionFactory.java

```diff
@@ -129,14 +138,35 @@ private HederaEvmTransaction fromHapiCreate(
                 body);
     }
 
-    private HederaEvmTransaction fromHapiCall(@NonNull final ContractCallTransactionBody body) {
-        throw new AssertionError("Not implemented");
+    private HederaEvmTransaction fromHapiCall(
+            @NonNull final AccountID payer, @NonNull final ContractCallTransactionBody body) {
+        assertValidCall(body);
+        return new HederaEvmTransaction(
+                payer,
+                null,
+                body.contractIDOrThrow(),
+                NOT_APPLICABLE,
+                body.functionParameters(),
+                null,
+                body.amount(),
+                body.gas(),
+                NOT_APPLICABLE,
+                NOT_APPLICABLE,
+                null);
     }
 
     private HederaEvmTransaction fromHapiEthereum(@NonNull final EthereumTransactionBody body) {
         throw new AssertionError("Not implemented");
     }
 
+    private void assertValidCall(@NonNull final ContractCallTransactionBody body) {
+        final var minGasLimit =
```

Comment:
Do we want to have the same check for create transactions so that we fail fast?

Reply:
Hmm it may make sense...we are trying to mimic mono-service until after differential testing tho. 🤔

Reply:
that's a good reason to not do it now - agreed.